### PR TITLE
Nikita Yaschenko, 2538

### PR DESCRIPTION
**Pull request by Nafanya**
Photos from Yandex's "Top", "Recent" and "Photos of the day" sections. Images are cached in LruCache and on disk. You can also download full-sized image and watch the image in browser. Images are scaleable.

### CODE REVIEW

File: app/src/main/java/ru/ifmo/md/extratask1/yfotki/SectionFragment.java [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+                        saveToPreferences(key, 1);
+                        getLoaderManager().restartLoader(LOADER_IMAGES, Bundle.EMPTY, SectionFragment.this);
+                    }
+                    break;
+                case Constants.STATE_ACTION_STARTED:
+                    mIsRefreshing = true;
+                    mRefreshLayout.setRefreshing(true);
+                    break;
+                case Constants.STATE_ACTION_ERROR:
+                    Toast.makeText(getActivity(), "Error during update", Toast.LENGTH_SHORT).show();
```
Comment (svasilinets) [inline app/src/main/java/ru/ifmo/md/extratask1/yfotki/SectionFragment.java:379]: UI string -> resources

File: app/src/main/java/ru/ifmo/md/extratask1/yfotki/SquareImageView.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package ru.ifmo.md.extratask1.yfotki;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.ImageView;
+
+/**
+ * Created by Nikita Yaschenko on 15.01.15.
+ */
+public class SquareImageView extends ImageView {
```
Comment (svasilinets) [inline app/src/main/java/ru/ifmo/md/extratask1/yfotki/SquareImageView.java:10]: I can't see the reason to create this class

**Comment by svasilinets** [comment]
- 1. It's quite easy to get an infinite progress on the main screen
- 2. No additional feature: install as wallpaper
- 4. No pagination at any screen
- 1. An infinite progress If there is no connection at the second screen
- 2. No caching of large images
- 1. An adapter in section shows for a few second an old image and change it for a new one. Looks ugly

Additional points:
- 2. scaleable images
- 4. viewPager with different sections (I consider this as a replacement for a pagination, but it is not a pagination. You can read here what was meant: https://github.com/IFMO-MobDev-2014/extratask1/issues/12)

Result: 20 - 11 + 6 = 15

**Comment by Nafanya** [comment]
@svasilinets Large images are cached since both thumbnails and large images are downloaded with ImageDownloader. Maybe it takes more time for them to load as they are bigger in size than small ones. (If we speak about persistent disk caching, but not ram).

**Pull request closed without merging**